celery/backends/base.py

```diff
@@ -647,11 +647,36 @@ def _forget(self, task_id):
 
     def _store_result(self, task_id, result, state,
                       traceback=None, request=None, **kwargs):
+
+        if state in self.READY_STATES:
+            date_done = datetime.datetime.utcnow()
+        else:
+            date_done = None
+
         meta = {
-            'status': state, 'result': result, 'traceback': traceback,
-            'children': self.current_task_children(request),
+            'status': state,
+            'result': result,
+            'traceback': traceback,
             'task_id': bytes_to_str(task_id),
+            'date_done': date_done,
         }
+
+        if request:
+            request_meta = {
+                # Some unit tests mock the properties, casting to string to
+                # avoid serialization errors
+                'children': self.current_task_children(request),
+                'name': str(getattr(request, 'task', None)),
```

Comment:
Consider using Python2/3 compatibility methods.

Reply:
Can you be more specific?  I turned on py2/py3 inspections and also loaded py2 and py3 virtual envs and didn't get any warnings on these lines.. ?

Reply:
This [library](https://github.com/celery/vine/blob/master/vine/five.py), also accessible through [kombu](https://github.com/celery/kombu/blob/master/kombu/five.py) provides the proper Python2/3 compatibility functions. [kombu.utils.encoding](https://github.com/celery/kombu/blob/master/kombu/utils/encoding.py) provides certain functions as well.